firmware/Cargo.toml

```diff
@@ -49,48 +53,71 @@ direct-boot = ["esp32c3-hal/direct-boot"]
 
 [dependencies]
 esp32c3-hal = { version = "0.3", features = [
-  "embassy",
-  "embassy-time-systick",
+  #"embassy",
+  #"embassy-time-systick",
+  #"direct-boot",
 ] }
 riscv-rt = "0.10"
 riscv = "0.10"
 esp-alloc = { version = "0.1", features = ["oom-handler"] }
 embedded-hal = "0.2"
+embedded-svc = { version = "0.23", default-features = false, features = [
+#  "nightly",
+] }
 
 # Embassy stuff
 embassy-futures = "0.1.0"
 embassy-executor = { git = "https://github.com/embassy-rs/embassy", rev = "2528f451387e6c7b27c3140cd87d47521d1971a2", features = [
-  "integrated-timers",
+  #"integrated-timers",
   "nightly",
 ] }
-embassy-time = { version = "0.1.0", features = [
-  "tick-hz-16_000_000",
-  "unstable-traits",
+#embassy-time = { version = "0.1.0", features = [
+#  "tick-hz-16_000_000",
+#  "unstable-traits",
+#] }
+
+# Wi-Fi
+esp-wifi = { git = "https://github.com/esp-rs/esp-wifi.git", rev = "34b5d0d", features = [
+  "esp32c3",
+  "wifi",
+  "embedded-svc",
+], optional = true }
+smoltcp = { version = "0.8", default-features = false, features = [
+  "async",
+  "defmt",
+#FIXME: alloc feature spits errors out because defmt isnt 
+#  "alloc",
+  "socket-raw",
+  "socket-udp",
+  "socket-tcp",
+  "socket-icmp",
+  "socket-dhcpv4",
+  "medium-ethernet",
+  "proto-dhcpv4",
+  "proto-igmp",
+  "proto-ipv4"
 ] }
 
 # Logging
 defmt = "0.3"
 defmt-rtt = { version = "0.4", optional = true }
 panic_defmt = { path = "crates/panic_defmt" }
 defmt_esp_println = { path = "crates/defmt_esp_println", optional = true }
+esp-backtrace = { git = "https://github.com/esp-rs/esp-backtrace.git", rev = "312f8ea", default-features = false, optional = true }
 
 # Peripherals
 mpu6050-dmp = { version = "0.2", optional = true }
-# esp-wifi doesn't seem to be updated for the latest ecosystem crates
-# esp-wifi = { git = "https://github.com/esp-rs/esp-wifi", rev = "e8adb3153f9757986b2a6af09cbfb622c485dcfc", features = [
-# esp-wifi = { path = "../../../External/esp-wifi", features = [
-#     "esp32c3",
-#     "wifi",
-# ] }
 
 # Other crates
+strum = { version = "0.24", default-features = false }
```

Comment:
I don't think you ever actually used this or `cfg-if`, did you? You should remove them